File: go/test/protocol/default_serializer.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

type DefaultSerializer struct {
	OutContentType        *string  `toml:"out-content-type"`
	AcceptContentTypes    []string `toml:"accept-content-types"`
	IndicateCharacterData bool     `toml:"indicate-character-data"`
	AllowCharacterData    bool     `toml:"allow-character-data"`
	FailDeserialization   bool     `toml:"fail-deserialization"`
}

func (serializer *DefaultSerializer) GetSerializer() TestCaseSerializer {
	return TestCaseSerializer{
		testCaseSerializer{
			OutContentType:        serializer.GetOutContentType(),
			AcceptContentTypes:    serializer.GetAcceptContentTypes(),
			IndicateCharacterData: serializer.GetIndicateCharacterData(),
			AllowCharacterData:    serializer.GetAllowCharacterData(),
			FailDeserialization:   serializer.GetFailDeserialization(),
		},
	}
}

func (serializer *DefaultSerializer) GetOutContentType() *string {
	if serializer.OutContentType == nil {
		return nil
	}

	outContentType := *serializer.OutContentType
	return &outContentType
}

func (serializer *DefaultSerializer) GetAcceptContentTypes() []string {
	acceptContentTypes := make([]string, len(serializer.AcceptContentTypes))
	copy(acceptContentTypes, serializer.AcceptContentTypes)
	return acceptContentTypes
}

func (serializer *DefaultSerializer) GetIndicateCharacterData() bool {
	return serializer.IndicateCharacterData
}

func (serializer *DefaultSerializer) GetAllowCharacterData() bool {
	return serializer.AllowCharacterData
}

func (serializer *DefaultSerializer) GetFailDeserialization() bool {
	return serializer.FailDeserialization
}

File: rust/azure_iot_operations_protocol/tests/metl/default_serializer.rs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use serde::Deserialize;

#[derive(Deserialize, Debug)]
pub struct DefaultSerializer {
    #[serde(rename = "out-content-type")]
    pub out_content_type: Option<String>,

    #[serde(rename = "accept-content-types")]
    pub accept_content_types: Option<Vec<String>>,

    #[serde(rename = "indicate-character-data")]
    pub indicate_character_data: Option<bool>,

    #[serde(rename = "allow-character-data")]
    pub allow_character_data: Option<bool>,

    #[serde(rename = "fail-deserialization")]
    pub fail_deserialization: Option<bool>,
}
